### DISCUSSION TITLE
installing nvm with znap

### DISCUSSION THREAD

**Discussion by eduardocereto** [marlonrichert/zsh-snap#174] [Q&A] [answered]
Here's what I have:

```
 export NVM_COMPLETION=true
 export NVM_LAZY_LOAD=true
 znap source lukechilds/zsh-nvm
```

This works fine without the `NVM_LAZY_LOAD` but when I set it to true I get:

```
_zsh_nvm_lazy_load:26: cmd: attempt to assign private in nested scope
```

Something weird related to how zsh-snap handles the sourcing of the script. I know znap has other features related to lazy loading. Any tips on how I can get it setup to lazy load nvm when needed?

**Comment by marlonrichert** [comment, marked as answer]
This happens [because of an annoying bug](https://www.zsh.org/mla/workers/2021/msg01569.html) in the [`zsh/param/private` module](https://zsh.sourceforge.io/Doc/Release/Zsh-Modules.html#The-zsh_002fparam_002fprivate-Module), for which (as far as I know) no fix has yet been submitted.

I will have to work around it by renaming my variables to something unlikely to be used by any plugins, which is [something I thought I already did](#109), but I it looks like I missed a spot (or introduced a new one).

**Reply by marlonrichert** [reply]
Should be fixed now: de2e7c684b0a2cf6b6f5a160927a74511475a515
